### PR TITLE
Noninterpretable loincs

### PR DESCRIPTION
**Pull request by aeyates**
This implements an additional tab-delimited file for storing information on LOINCs that should not be converted to HPO terms. Optionally the row can contain a reason it should not be interpreted. When a conversion of one of these LOINCs is requested, it will fail with a NonInterpretableLoincException. This will distinguish it from LoincNotAnnotatedException.

### CODE REVIEW

File: src/main/resources/noninterpretable-annotations.tsv [hunk 1]
```diff
@@ -0,0 +1 @@
+loincId	loincScale	reason1752-5	Qn	Quality control parameter55284-4	-	Components not mapped yet
```
Comment (aeyates) [inline src/main/resources/noninterpretable-annotations.tsv]: The second LOINC in this file, 55284-4, is for Blood Pressure, and ultimately should be annotated once we figure out components. I added it only for convenience in testing as it is encountered often in sandboxes.

Comment (kingmanzhang) [reply src/main/resources/noninterpretable-annotations.tsv]: This design is pretty good. Currently we have one list of LOINC codes for tests with "unspecified specimen" (https://github.com/TheJacksonLaboratory/loinc2hpoAnnotation/blob/develop/Data/LOINC%20CATEGORY/unspecified_specimen.txt) and one list of LOINC panels (along with components). 
https://github.com/TheJacksonLaboratory/loinc2hpoAnnotation/tree/develop/Data/LoincPanel
Peter thinks tests with unspecified specimen should not be mapped, so we can use the list to exclude such FHIR messages. Many LOINC panels (and their components) should not be mapped either. We can discuss the details in a meeting, but briefly, we will annotate in the panel file whether a LOINC panel can be mapped to HPO (e.g. whole blood count panel cannot), and for the components of each panel, whether a component is required for a test(already annotated by Regenstrief), is required for mapping parent to a HPO term and whether itself can be mapped to an HPO term. Refer to the lines for LOINC panel 35094-2 and 35088-4 for example.

Comment (kingmanzhang) [reply src/main/resources/noninterpretable-annotations.tsv]: consider the LOINC panel annotation to be a future project; for now, I think the key is to implement a few examples for demo purposes.

Comment (kingmanzhang) [reply src/main/resources/noninterpretable-annotations.tsv]: How did you find whether a LOINC is for QC? Is it possible to export them to a list all at once?

Comment (aeyates) [reply src/main/resources/noninterpretable-annotations.tsv]: I got the first LOINC from the issue Peter pointed us to where he mentions that it is a quality control parameter: https://github.com/obophenotype/human-phenotype-ontology/issues/3513. I don't have the clinical experience to make those determinations but wanted to demonstrate that if having a "reason" was useful we could do it. We could also leave it blank or put "Unspecified Specimen" for the list you provided.

**Comment by pnrobinson** [comment]
I would agree that it is good to denote that this is used as a Q/C parameter in the file. It might become useful later on...

**Comment by aeyates** [comment]
I added the unspecified_specimens list to the noninterpretable-annotations file. I used the LOINC scale "Qn" for all annotations rather than look them up, since our algorithm doesn't use this. Will need to fix this later. I ran through all the sandboxes, but it didn't change statistics much since most of the LOINCs are not exercised.

**Pull request merged**